### PR TITLE
Rebase theme to a different baseurl

### PR DESCRIPTION
**Pull request by trenki2**
I want to use this theme on a github project page. But it seemed to only work when everything was at the root /. I set baseurl in the config and modified the site, so that all relevant url prepend the baseurl. It works not with jekyll serve.

### CODE REVIEW

**Comment by daviddarnes** [comment]
Hi @trenki2, thanks for contributing! To be honest I've been avoiding this problem as I didn't see a very elegant way to solve it, however this may just work. I'm going to test it and get back to you.

This PR relates to #40 and #42

File: _includes/site-search.html [hunk 1]
```diff
@@ -32,7 +32,7 @@
           return `
               <li class="item  item--result">
                   <article class="article  typeset">
-                      <h4><a href="${item.url}" title="Result - ${item.title}">${item.title}</a></h4>
+                      <h4><a href="{{ site.baseurl }}${item.url}" title="Result - ${item.title}">${item.title}</a></h4>
```
Comment (daviddarnes) [inline _includes/site-search.html]: You can remove `{{ site.baseurl }}` as that comes from the json file

File: _config.yml [hunk 2]
```diff
@@ -87,6 +87,7 @@ title: "Alembic" # Site name or title, also used in jekyll-seo-tag
 logo: "/assets/logo.svg" # Site logo, also used in jekyll-seo-tag
 description: "Alembic is a starting point for Jekyll projects. Rather than starting from scratch, this boilerplate is designed to get the ball rolling immediately" # Site description and default description, also used in jekyll-seo-tag
 url: "https://alembic.darn.es" # Site url, also used in jekyll-seo-tag
+baseurl: "/baseurl"
```
Comment (daviddarnes) [inline _config.yml]: I test with this option turned on and it worked well, however I'd like to have the option empty in the main repo. Can you change it to just `baseurl: ""`, thanks

**Comment by trenki2** [comment]
I have still found some issues after I made this pull request. I commited and pushed them to my fork. They are available there.

**Comment by daviddarnes** [comment]
@trenki2 ah crap, I missed that one. Good spot! I'll take a brief look but it should be fine

File: _config.yml (hunk 2 above)
Comment (daviddarnes) [inline _config.yml]: This is the only change left I think, then I'm totally merging it in 👍

**Comment by daviddarnes** [comment]
Done, thank you @trenki2

**Pull request merged**